src/main/java/io/confluent/kafka/connect/datagen/DatagenConnectorConfig.java

```diff
@@ -90,7 +93,17 @@ public static ConfigDef conf() {
           Importance.HIGH,
           QUICKSTART_DOC
         )
-        .define(RANDOM_SEED_CONF, Type.LONG, null, Importance.LOW, RANDOM_SEED_DOC);
+        .define(RANDOM_SEED_CONF,
+          Type.LONG,
+          null,
+          Importance.LOW,
+          RANDOM_SEED_DOC)
+        .define(GENERATE_TIMEOUT_CONF,
+          Type.LONG,
```

Comment:
> For backward compatibility of already deployed connectors which don't use this config value, its better to leave default as null.

If there is a default value, say `1000`, it will get applied to already deployed connectors on upgrades just fine.

> Why don't we use ConfigDef.NO_DEFAULT_VALUE here?

That implies that the config has no default value and must be specified, which would actually be backward incompatible and existing connectors would fail on upgrades.

Reply:
> If there is a default value, say `1000`, it will get applied to already deployed connectors on upgrades just fine.
> 
> >

But what if some on-prem customer has a regex which takes more time than this default value? It would be fine for cloud since we won't expose this in cloud.

Reply:
so, let's go ahead with null default value in that case! for existing connectors, we can retain existing behavior, unless the config is explicitly specified.

Reply:
@garrix-fan that makes sense, I was just trying to say that we can't use no default value - `null` is fine.